### PR TITLE
Integrate refactorings

### PR DESCRIPTION
**Pull request by xeolabs**

### CODE REVIEW

File: src/viewer/Viewer.js [hunk 1]
```diff
@@ -422,80 +427,94 @@ class Viewer {
         // canvas ourselves, in order to allow the Viewer to render the
         // right amount of pixels, for a sharper image.
 
+        return new Promise((resolve, reject) => {
```
Comment (Amoki) [inline src/viewer/Viewer.js:430]: rollback of https://github.com/xeokit/xeokit-sdk/pull/1118 ?

File: src/plugins/XKTLoaderPlugin/parsers/ParserV10.js [hunk 2]
```diff
@@ -429,12 +429,6 @@ function load(viewer, options, inflatedData, sceneModel) {
                                 geometryArrays.geometryIndices = indices.subarray(eachGeometryIndicesPortion [geometryIndex], atLastGeometry ? indices.length : eachGeometryIndicesPortion [geometryIndex + 1]);
                                 geometryValid = (geometryArrays.geometryPositions.length > 0 && geometryArrays.geometryIndices.length > 0);
                                 break;
-                            case 4:
```
Comment (Amoki) [inline src/plugins/XKTLoaderPlugin/parsers/ParserV10.js:432]: rollback of https://github.com/xeokit/xeokit-sdk/pull/1114 ?

**Comment by Amoki** [comment]
@xeolabs I fear you reverted some PR in the merge, I don't know if it's voluntary.

**Comment by xeolabs** [comment]
Yes involuntary - will restore this evening, no fear

**Pull request merged**